server/gendb/tests.go.tmpl

```diff
@@ -453,34 +453,6 @@ func Get{{ $modelName }}sBy{{ $indexName }}(d db.Interface, t *testing.T) func(t
 {{- end }}
 
 {{- if $xdbConfig.AllowPrimaryIndexScan }}
-type scan{{ $modelName }}sInput struct {
-  ctx   context.Context
-  input db.Scan{{ $modelName }}sInput
-}
-type scan{{ $modelName }}sOutput struct {
-  {{ varname $modelName }}s []models.{{ $modelName }}
-  err    error
-}
-type scan{{ $modelName }}sTest struct {
-  testName string
-  d        db.Interface
-  input    scan{{ $modelName }}sInput
-  output   scan{{ $modelName }}sOutput
-}
-
-func (g scan{{ $modelName }}sTest) run(t *testing.T) {
-  {{ varname $modelName }}s := []models.{{ $modelName }}{}
-  err := g.d.Scan{{ $modelName }}s(g.input.ctx, g.input.input, func(m *models.{{ $modelName }}, last bool) bool {
-    {{ varname $modelName }}s = append({{ varname $modelName }}s, *m)
-    return true
-  })
-  var errStr string
-  if err != nil {
-    errStr = err.Error()
-  }
-  require.Equal(t, g.output.err, err, errStr)
-  require.Equal(t, g.output.{{ varname $modelName }}s, {{ varname $modelName }}s)
```

Comment:
Why change the whole format of how these tests are written, rather than just update these assertions?

Reply:
I was able to get the basic scan test working in the old format by just changing the assertion to not care about the order of the results from the scan. The basic scan tests return all 3 items in the db, so we can use `assert.ElementsMatch` to assert that the result has the same items as what's expected in any order https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch.

However, I wasn't sure how to do that with the tests for starting after. Since we don't know how the items are ordered in the db, if we scan for items starting after x, we don't know how many items there after after x since x can be the first, last, or anything in between. The way I did it was to find the first item, and then scan with starting after the first item so that we know how many items to expect.

Any suggestions on how we could make it work in the existing test format?

Reply:
Ah I see, that makes sense. Yeah I don't really have a better idea.

Also since all of these only generate 1 test case each, table tests aren't actually that helpful/different, so there's actually not much benefit to using that structure now that I think about it more carefullly

Reply:
I can add in a comment in the codegen template to provide some context as to why the scan test is structured differently than everything else.